ostro-os: update to latest HEAD

avalluri opened the pull request:
test-build: Update ostro-os to latest HEAD (as per @mythi request)

Signed-off-by: Amarnath Valluri amarnath.valluri@intel.com

avalluri committed:
ostro-os: update to ostro-os master

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
meta-ostro-xt/upower: Rename upower bbappend

In order to allow meta-oe layer updates, rename upower bbappend with more
generic version.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
perf: Remove runtime dependency on "man"

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
librealsense: Fix package QA warnings: 'No GNU_HASH in the elf binary:'

Patched source to respect LDFLAGS.

Patch submitted to meta-intel-relasense layer:
https://github.com/IntelRealSense/meta-intel-realsense/pull/4

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
glmark2: Fix build errors by enabling c++ 11 features

When build with gcc5, build was failing with errors:
| ../src/image-reader.cpp: In member function 'bool PNGReader::init(const
string&)':
| ../src/image-reader.cpp:123:16: error: 'unique_ptr' in namespace 'std' does
not name a template type
|      const std::unique_ptr<std::istream>
is_ptr(Util::get_resource(filename));
|                 ^
| ../src/image-reader.cpp:124:12: error: 'is_ptr' was not declared in this
scope
|      if (!(*is_ptr)) {
|             ^
| ../src/image-reader.cpp:149:57: error: 'is_ptr' was not declared in this
scope
|      png_set_read_fn(priv_->png, reinterpret_cast<void*>(is_ptr.get()),
|                                                          ^
| ../src/model.cpp: In member function 'bool Model::load_3ds(const string&)':
| ../src/model.cpp:364:16: error: 'unique_ptr' in namespace 'std' does not name
a template type
|      const std::unique_ptr<std::istream>
input_file_ptr(Util::get_resource(filename));
|                 ^
| ../src/model.cpp:365:31: error: 'input_file_ptr' was not declared in this
scope
|      std::istream& input_file(*input_file_ptr);
|                                ^
| ../src/model.cpp: In member function 'bool Model::load_obj(const string&)':
| ../src/model.cpp:684:16: error: 'unique_ptr' in namespace 'std' does not name
a template type
|      const std::unique_ptr<std::istream>
input_file_ptr(Util::get_resource(filename));
|                 ^
| ../src/model.cpp:685:30: error: 'input_file_ptr' was not declared in this
scope
|      std::istream& inputFile(*input_file_ptr);

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

mythi commented on meta-ostro-xt/recipes-devtools/checksec/checksec_1.5-1.bbappend:
```diff
@@ -0,0 +1,3 @@
+SRC_URI[md5sum] = "075996be339ab16ad7b94d6de3ee07bd"
+SRC_URI[sha256sum] = "77b8a7fd9393d10def665658a41176ee745d5c7969a4a0f43cefcc8a4cd90947"
```
Is this still wrong? IIRC 1.5-1 should fix that.

avalluri replied:
@mythi  Yes, bitbake complained abt the mismatch.

ERROR: checksec-1.5-1-r0 do_fetch: Checksum failure fetching http://www.trapkit.de/tools/checksec.sh
ERROR: checksec-1.5-1-r0 do_fetch: Function failed: Fetcher failure for URL: 'http://www.trapkit.de/tools/checksec.sh'. Checksum mismatch!
File: '/var/tmp/ostro-os-xt/bb-cache/sources/checksec.sh' has md5 checksum 075996be339ab16ad7b94d6de3ee07bd when 57cc3fbbbe48e8ebd4672c569954374d was expected
File: '/var/tmp/ostro-os-xt/bb-cache/sources/checksec.sh' has sha256 checksum 77b8a7fd9393d10def665658a41176ee745d5c7969a4a0f43cefcc8a4cd90947 when 05822cd8668589038d20650faa0e56f740911d8ad06f7005b3d12a5c76591b90 was expected
If this change is expected (e.g. you have upgraded to a new version without updating the checksums) then you can use these lines within the recipe:
SRC_URI[md5sum] = "075996be339ab16ad7b94d6de3ee07bd"
SRC_URI[sha256sum] = "77b8a7fd9393d10def665658a41176ee745d5c7969a4a0f43cefcc8a4cd90947"

mythi replied:
In this case the .sh comes from the source mirror it seems. We should probably get rid of that mirrored copy and fix ostro-os-xt master with: https://github.com/ostroproject/ostro-os/commit/913b474c47dd83e2d4ea564464c8e21186867dd7

avalluri replied:
@mythi How can we get rid of the mirror?, you mean, should we touch the recipe version?

mythi replied:
@avalluri yes we need to take checksec_1.5-1.bb in ostro-os-xt. (Then in this PR you revert the commit (because the same .bb comes from ostro-os).)

avalluri replied:
@mythi, With this PR ostro-os-xt is using checksec_1.5-1.bb which is coming from ostro-os only.

I tried re-build checksec  without this patch by disabling source mirror(setting OSTRO_SOURCE_MIRROR_URL in local.conf), this way force fetched from the upstream, which resolved the mismatch issue. Hence reverting my commit.

But i am not sure, how it will be resolved at CI environment.

mythi replied:
@avalluri we need to get checksec_1.5-1.bb in ostro-os-xt because everybody building the distro without OSTRO_SOURCE_MIRROR_URL gets the failure. The old copy of checksec.sh only exists in our mirror.

avalluri replied:
@mythi Anyway I am patching the checkseck recipe to avoid this mirroring issue, hopefully should fix this.

avalluri replied:
@mythi  In your opinion does this makes any sense: 
https://github.com/avalluri/meta-security-isafw/commit/cd0cc55723918f22e77a47df7741278591972f79

mythi commented:
On Mon, Sep 19, 2016 at 2:47 PM, Amarnath Valluri notifications@github.com
wrote:

> ## _@avalluri_ commented on this pull request.
> 
> In meta-ostro-xt/recipes-devtools/checksec/checksec_1.5-1.bbappend
> https://github.com/ostroproject/ostro-os-xt/pull/20:
> 
> > @@ -0,0 +1,3 @@
> > +SRC_URI[md5sum] = "075996be339ab16ad7b94d6de3ee07bd"
> > +SRC_URI[sha256sum] = "77b8a7fd9393d10def665658a41176ee745d5c7969a4a0f43cefcc8a4cd90947"
> 
> @mythi https://github.com/mythi In your opinion does this makes any
> sense:
> avalluri/meta-security-isafw@cd0cc55
> https://github.com/avalluri/meta-security-isafw/commit/cd0cc55723918f22e77a47df7741278591972f79
> 
> It looks fine but we'd anyway need to fix the build for ostro-os-xt. With
> the fix avalluri/meta-security-isafw@cd0cc55
> would not be needed.

avalluri committed:
tinymembench: Fix QA Issue: No GNU_HASH in the elf binary

Patched Makefile to make use of LDFLAGS

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
gstreamer-vaapi-1.0: Fallaback to v0.7.0

As latest(v1.8.0) gstreamer-vaapi is having hard dependency on gstreamer-codecs
which are part of bad plugins, we are falling back to old version.

This is a temporary fix.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
backport-iwlwifi: Fix random build failure

Randomly build fails with below error:

   Generating local configuration database from kernel ...make[1]: ***
   No rule to make target 'kernelversion'.  Stop.
   Kernel version parse failed!
   make: *** [Makefile:42: defconfig-iwlwifi-public] Error 1

The reason for this failure is 'make defconfig-iwlwifi-public' in do_configure
task trys to get 'kernelverison' from kernel scripts. Kernel scripts are
generated by 'do_make_scripts'. The do_make_script is run before running
do_compile. So at the time of do_configure, kernel sripts are not ready,
hence 'make kernelversion' fails.

This change moves 'make defconfig-iwlwifi-public' to do_comiple.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
ocl-icd: Disable manpage generation

Manpage generation using(a2x) xmllint, which needs network to download custom xml DTD.
Fails with below error:

| a2x: executing: "xmllint" --nonet --noout --valid "/home/avalluri/ostro-os-xt/build/tmp-glibc/work/corei7-64-ostro-linux/ocl-icd/2.2.9-r0/ocl-icd-2.2.9/doc/libOpenCL.7.xml"
|
| I/O error : Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd
| /home/avalluri/ostro-os-xt/build/tmp-glibc/work/corei7-64-ostro-linux/ocl-icd/2.2.9-r0/ocl-icd-2.2.9/doc/libOpenCL.7.xml:2: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
| D DocBook XML V4.5//EN" "http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd"
|                                                                                ^

Patched configure.ac to add config option to enable/disable document generation.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

kad commented on docker/build-project.sh:
```diff
@@ -25,6 +25,12 @@ cd $WORKSPACE
+git config --global user.name "$GIT_COMMITTER_NAME"
+git config --global user.email "$GIT_COMMITTER_EMAIL"
+
 # FIXME: undbound variables used without checking:
```
Don't do that. Patches must have origin information. This will hide that error situation.

avalluri replied:
@Kad, This is not just related to git apply or patch formatting, As i mentioned in commit message the real issue is with the git environment in bitbake tasks, though we set GIT_COMITTER_{NAME/EMAIL},  those are not available while patching source. So here the idea is to pass them via 'git config'

kad replied:
If patch in your changeset would have proper information in header, those wouldn't be needed.

avalluri replied:
@kad, This issue raised on yocto project by @mythi and they accepted to fix in oe-core. Hopefully we need not to set this configuration as oecore itself provide dummy git config.

https://bugzilla.yoctoproject.org/show_bug.cgi?id=10346

We can revert this once the fix is available.

kad replied:
let's get fix from oe-core (Paul's change to patch functions). it's better than trying to create local workarounds.

avalluri replied:
Paul's fix your are referring(http://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=4b4387455c62cf19fa3d215a11b5d0b1211d4570) does not fix kernel patching issue, that address only PATCHTOOL='git' case.

This is the fix we needed:
http://lists.openembedded.org/pipermail/openembedded-core/2016-October/127230.html

kad replied:
in reality, I think both of those patches are needed. we stepped on one case and issue present in different scenarios.

kad commented on docker/local-build.sh:
```diff
@@ -11,9 +11,12 @@ BUILD_CACHE_DIR=$WORKSPACE/bb-cache
 TARGET_MACHINE="intel-corei7-64"
+GIT_COMMITTER_EMAIL=${GIT_COMMITTER_EMAIL:-$(git config --get user.email)}
+GIT_COMMITTER_NAME=${GIT_COMMITTER_NAME:-$(git config --get user.name)}
+GIT_COMMITTER_NAME=${GIT_COMMITTER_NAME:-${USER}}
```
Don't do that. Git apply error is real error. Fix it, instead of trying to hide.

kad commented on docker/local-build.sh:
```diff
@@ -11,9 +11,12 @@ BUILD_CACHE_DIR=$WORKSPACE/bb-cache
 
 BUILD_ARGS="--build-arg uid=`id -u`"
-RUN_ARGS="-u `id -u`"
+RUN_ARGS=(-u "`id -u`")
```
Why is that change ?

kad commented on docker/local-build.sh:
```diff
@@ -22,7 +25,7 @@ for proxy in $safe_proxy_vars; do
         val="`echo ${!proxy} | tr -d ' '`"
         BUILD_ARGS="$BUILD_ARGS --build-arg $proxy=${val}"
-        RUN_ARGS="$RUN_ARGS -e $proxy=${!proxy}"
+        RUN_ARGS+=(-e "$proxy=${!proxy}")
```
why is that here ?

kad commented on docker/local-build.sh:
```diff
@@ -39,19 +42,19 @@ echo "$BUILD_NUMBER" > $WORKSPACE/.build_number
 fi
 
-docker run -it --rm $RUN_ARGS \
+docker run -it --rm "${RUN_ARGS[@]}" \
```
why to have this bashism ?

avalluri replied:
@kad, Current script fails to run, if any of the docker run environment we pass has white spaces in it. I tried to solve this by using bash arrays.

kad replied:
any space in this subset of variables usually means syntax error, which would be visible if not in this script, but then further down the line. Better sanitize values near line 25.

avalluri replied:
This may not be the case for GIT_COMITTER_NAME(though its still open :)), where space is valid. That is what actually made visible this issue.
The other possibility where space is valid is, passing {build_target(s)} to the script via command line arguments, to build custom targets(this is how i am using this script with my local changes). This way we can use all bitbake options with this script.

avalluri replied:
@kad As missing git config issue is been fixed in oe-core itself, I feel we can live without this change. The main motivation behind this change was passing GIT_COMITTER_NAME to docker, where space is valid.

kad replied:
GIT_\* variables I don't think it's really good idea to pass. Same with *_proxy variables: better to sanitize  content of *_proxy variables.
However, if you split patch for making RUN_ARGS as array and test it under different versions of bash (3 and 4 at least), I think it can be accepted.
Another potential enhancement that can be done in build scripts is to parse BB_ENV_EXTRAWHITE and pass those as arguments to docker calls. that way user would have explicit way to pass something from his development environment into isolated container.

avalluri replied:
@kad, I couldn't understand fully, about splitting this patch. This patch is touching only RUN_ARGS.

kad replied:
I mean, put only this patch that touches RUN_ARGS into separate PR.

avalluri replied:
@kad, I put this commit in PR #44
As you suggested, I tested this change both on bash v4.2 & v3.2. But our current script does not work with v3.2 because 'test/if' in bash-v3 does not support '-v' option, other than that, this change work on both versions.

kad replied:
Hm. interesting. that's something that also might need to be fixed. I'll check. Let's continue discussion about that part in #44

kad commented on meta-ostro-xt/recipes-opencl/ocl-icd/ocl-icd/configurable-doc-generation.patch:
```diff
@@ -0,0 +1,25 @@
+diff --git a/configure.ac b/configure.ac
```
This patch must have proper origin header, to clearly specify where it comes from, who is author, upstream status, etc.

avalluri replied:
@kad, Thanks for pointing out, I was in impression that git formatted patch fail to apply on non-git source, now I git formatted the patch.

mythi replied:
@avalluri what were the other patches git apply failed and f089278 was needed?

avalluri replied:
@mythi, Currently there are two patches in kernel patch series on XT:
1. librealsense_formats_linux-yocto_4.4.patch
2. 0001-ovl-setxattr-don-t-deadlock-when-called-from-ima_fix.patch

Both fail to apply if we not provide committer details either via environment or via global git config.
This is not visible on non-docker builds as, usually host .gitconfig set properly, But not the case with docker container.

mythi replied:
Thanks
1. we have reported to meta-intel-realsense. 2. is ours but it looks correct. Any idea why it fails?

avalluri replied:
Almost all git actions(atleast in this context am,apply) look for committer details before applying the patch. That's what i observed with this exercise :)

kad replied:
that's the reason for clean build inside container: catch inconsistencies in data and patches :)

kad reviewed (changes requested):
Fix errors in new patches. Don't try to hide them.

avalluri committed:
ostro-xt-supported-recipes: New reicpes to support SVG image

Commit 920baaa6eb4c5094b7fe36736c598badf43350dc on oe-core added 'libsrvg-gtk'
dependency to packagegroup-xfce-base.bb, to support viewing of svg images.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

okartau commented:
test this please

avalluri committed:
linux-firmware_git: Drop the local change

The changes of this bbappend have been applied on oe-core. Hence this change is
not needed anymore.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri committed:
ant-native: Fix random build failure

Due to lack of task dependency between 'do_removebinaries' and 'do_removecruft'
when both run parallelly sometimes, 'find' in do_removebinaries referring on directories
that sare already removed by do_removecruft.

This change makes sure that both tasks execute sequentially.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri commented:
@kad As you suggested made a new PR for build script change: https://github.com/ostroproject/ostro-os-xt/pull/44

avalluri committed:
linux-yocto: fix do_patch failure due to lack of git config

This issue been addressed in oecore here:
http://lists.openembedded.org/pipermail/openembedded-core/2016-October/127230.html

Till we take 25b43cb05c645e43f96bc18906441b8fdc272228 from oecore, this is a
temporary fix to whitelist and pass git committer environment to do_patch task.

Signed-off-by: Amarnath Valluri <amarnath.valluri@intel.com>

avalluri commented:
@kad Now i reverted the change(setting git config inside the docker) you are not happy with :) can you please review if the new change (https://github.com/ostroproject/ostro-os-xt/pull/20/commits/79026c755a86d739fa667fa410090bbc10644805) is ok.

okartau commented:
test this please
